### ISSUE TITLE
What happen to mumble-docker

### ISSUE THREAD

**Issue by jobber2955** [mumble-voip/mumble#5613]
### The issue

I see [mumble-docker](https://github.com/mumble-voip/mumble-docker) repo has been ```takedown``` for DMCA problem, but it has 28 stars...

What happened? Why a docker image will be  ```takedown``` by DMCA... Will ```mumble-docker``` be available soon?

### Mumble version

1.4.0

### Mumble component

Server

### OS

Linux

**Comment by Krzmbrzl** [comment]
See https://github.com/sudoforge/docker-mumble-server/issues/133

**Comment by jobber2955** [comment]
Thanks, hope everything goes well and thanks for replaying

**Comment by jobber2955** [comment]
@Krzmbrzl sorry for re-open&comment this issues, but could you privately send me a copy of ```mumble-docker``` repo since I don't know when and whether ```mumble-voip/mumble-docker``` will be available again. And BTW, will ```mumble-voip/mumble-docker``` be available again? will you give me a rough time that we needs to wait? 

##### I'm acting so hurry no because I need this docker image, I could just ```apt-get install``` it, but I'm gald to see and hope mumble become more and more modern and advanced.

I tried directly pull image from docker hub but couldn't find a way to custom the config .ini file. So it would be nice to have a repo and custom my own .ini file, then docker build it myself.

To be honest, I also tried build mumble-server and the docker image myself, it was success, but I'm not familiar with docker, so after my ```docker build```, the final image is so huge that could not be an option. So if a copy of ```mumble-voip/mumble-docker``` is not workable, I will try to learn more about docker and build it myself.

**Comment by Krzmbrzl** [comment]
I'm pretty sure that I am legally not allowed to pass the repo along to you. However, we are still working on bringing the repo back up again and as it stands right now, this should either happen tomorrow or the day after tomorrow.

Meanwhile, as you have figured out already, you can still pull the Docker image from https://hub.docker.com/r/mumblevoip/mumble-server.
In terms of configuration, you can actually have a look at https://github.com/sudoforge/docker-mumble-server/blob/trunk/README.md as the configuration of the now taken down version of the Dockerfile (which is the one that was used to build the image on Dockerhub) is virtually identical to this one. Just use the environment variables to customize the used config options.

**Comment by jobber2955** [comment]
@Krzmbrzl thanks!

**Comment by theAkito** [comment]
I still don't understand why that guy issued a DMCA claim, just because parts of his open source project, that only exists because of this one, was copied for the original project, i.e. this one. Am I the only one who thinks that guy's stupid?

**Comment by Krzmbrzl** [comment]
I understand thst he is/was upset about the copying without attributing the original author. Personally, I would have considered it enough for that attribution to be added to the most recent state of the repo, but obviously he has a different point of view here.

**Comment by theAkito** [comment]
> I understand that he is/was upset about the copying without attributing the original author.

I mean, I get that part, in general. But in this case, the maintainers of the original project he provides a Docker image for are copying his work, so it's not randomly used by some malicious entity, but by the actual authors of the software, which defines his Docker image repository's whole existence. Without `murmur` his Docker repository wouldn't even exist in the first place.

Well, I would be honoured, if my work were re-used by the maintainers of the original project. Okay, I can understand the request for attribution, but I wouldn't ever file a DMCA claim. I would probably just open a pull request, adding my signature. That's it.

DMCA just seems like such a drastic measure in this specific situation.

As of now, I do not know which parts were copied and how big/complicated they were. But in my experience, with small instruction sets or languages like Dockerfile, it's hard not to copy someone out there. It's just such a simple way of defining a software deployment, you will eventually run into "copied" code, without having copied anything.

**Comment by Krzmbrzl** [comment]
Yeah, well :shrug:

There's no real point in discussing this here though as the DMCA has already been filed, so we now have to focus on getting this resolved in order to have the repo up and running again...

**Comment by davidebeatrici** [comment]
For reference: https://github.com/github/dmca/blob/master/2022/03/2022-03-31-docker-mumble-server.md

**Comment by jdaviescoates** [comment]
> we now have to focus on getting this resolved in order to have the repo up and running again...

Where is the best place to keep track of that @Krzmbrzl ? Or do we just keep refreshing https://github.com/mumble-voip/mumble-docker until if comes back? :)

**Comment by Krzmbrzl** [comment]
Uhm... I guess this issue is the best place for that. I'll reopen it until we resolved the issue.

**Comment by gclawes** [comment]
What's the difference between the container that was removed and this? https://github.com/mumble-voip/mumble/blob/master/Dockerfile

**Comment by jobber2955** [comment]
@gclawes Don't know but I think there is no big different.

For now, I wrote my own ```Dockerfile``` and build the image. After having a look at the ```Dockerfile``` at this repo, I think there is no actual change & different.

You could just custom your own ```mumble-server.ini```, and in the ```Dockerfile``` just ```COPY``` it into the image.

Or you could do the same thing as me: Build the mumble-server executable, then just write a simple ```Dockerfile``` then ```COPY``` executable and config file and SSL files, finally write a simple script to set a superuser password and then start the server.(If you don't need a superuser password, you could just write ```CMD``` command in the ```Dockerfile```)

**Comment by Krzmbrzl** [comment]
The Dockerfile itself is quite similar, but the difference is the entrypoint. The dedicated repo supports multiple options that can be specified to change the behavior of the running mumble server.

**Comment by gclawes** [comment]
Right.  I implemented the original version of the one in this repo to, it expected a murmur.ini to be mounted in to change it's behavior.

Looking at `sudoforge/docker-mumble-server` and assuming it's entrypoint is the same/similar as the one removed, I see how it's more convenient to pass in environment variables that can be rendered into a `murmur.ini` (though it does look like it would log sensitive values like DB passwords this way).  

However, if the dedicated repo in this project isn't coming back with this entrypoint due to DMCA, would it be so bad to start publishing images based on the [Dockerfile](https://github.com/mumble-voip/mumble/blob/master/Dockerfile) in this repo?

**Comment by Krzmbrzl** [comment]
> though it does look like it would log sensitive values like DB passwords this way

It does indeed, but if you don't specify the password via command-line, the server itself will spit out the auto-generated to the log anyway, so the password is essentially always logged (when the server is freshly created).

> if the dedicated repo in this project isn't coming back with this entrypoint due to DMCA

It will come back up. We have already prepared a new version that is pretty much ready for shipping, but we wanted to get an explicit confirmation from GitHub that the DMCA has been resolved. However, I get the feeling that this is never going to happen as they were very evasive when I asked for clarification on the topic :eyes:

**Comment by danger89** [comment]
This is crazy. too bad that guy didn't help the open source community. bleh

**Comment by Krzmbrzl** [comment]
Since I gave up hope that we will ever receive definitive confirmation of GitHub that the DMCA has been resolved, we just published the new version of the repo that we have been working on in the background. It can be found again at https://github.com/mumble-voip/mumble-docker

Note that the image on Dockerhub has not yet been updated yet and the images are not compatible in terms of the used configuration options.
We'll polish the release workflow and then we can push the new version to Dockerhub as well.
